unblob/handlers/compression/lz4.py

```diff
@@ -0,0 +1,176 @@
+        outfile = Path(inpath).stem
+        return ["lz4", "--decompress", inpath, f"{outdir}/{outfile}"]
+
+
+class LegacyFrameHandler(_LZ4HandlerBase):
+    NAME = "lz4_legacy"
+    YARA_RULE = r"""
+        strings:
+            $lz4_legacy_frame = { 02 21 4C 18 }
+        condition:
+            $lz4_legacy_frame
+    """
+
+    def calculate_chunk(  # noqa: C901
+        self, file: io.BufferedIOBase, start_offset: int
+    ) -> Optional[ValidChunk]:
+
+        file.read(4)  # skip magic bytes
+
+        while True:
+            # The last block is detected either because it is followed by the “EOF” (End of File) mark,
+            # or because it is followed by a known Frame Magic Number.
+            raw_bsize = file.read(4)
+            if raw_bsize == b"":  # EOF
+                break
+
+            block_compressed_size = struct.unpack("<I", raw_bsize)[0]
+            if block_compressed_size in FRAME_MAGICS:
+                # next magic, read too far
+                file.seek(-4, io.SEEK_CUR)
+                break
+
+            file.read(block_compressed_size)
+
+        end_offset = file.tell()
+        return ValidChunk(start_offset=start_offset, end_offset=end_offset)
+
+
+class SkippableFrameHandler(_LZ4HandlerBase):
+    NAME = "lz4_skippable"
+    YARA_RULE = r"""
+        strings:
+            $lz4_skippable_frame = { 5? 2A 4D 18 }
+        condition:
+            $lz4_skippable_frame
+    """
+
+    def calculate_chunk(  # noqa: C901
+        self, file: io.BufferedIOBase, start_offset: int
+    ) -> Optional[ValidChunk]:
+        file.read(4)  # skip magic bytes
+        frame_size = struct.unpack("<I", file.read(4))[0]
+        file.read(frame_size)
+        end_offset = file.tell()
+        return ValidChunk(start_offset=start_offset, end_offset=end_offset)
+
+
+class DefaultFrameHandler(_LZ4HandlerBase):
+    NAME = "lz4_default"
+
+    YARA_RULE = r"""
+        strings:
+            $lz4_default_frame = { 04 22 4D 18 }
+
+        condition:
+            $lz4_default_frame
+    """
+
+    def parse_flg(self, raw_flg: int):
+        flg = FLG()
+        flg.version = (raw_flg >> 6) & _2BITS
+        flg.block_independence = (raw_flg >> 5) & _1BIT
+        flg.block_checksum = (raw_flg >> 4) & _1BIT
+        flg.content_size = (raw_flg >> 3) & _1BIT
+        flg.content_checksum = (raw_flg >> 2) & _1BIT
+        flg.dictid = raw_flg & _1BIT
+        return flg
+
+    # block checksum: lz4 -BX lorem.txt lorem.blockchecksum.lz4
+    # content size: lz4 --content-size lorem.txt out.lz4
+    # no c size: lz4 --no-content-size lorem.txt out.lz4
+    # legacy:
+
+    def calculate_chunk(  # noqa: C901
+        self, file: io.BufferedIOBase, start_offset: int
+    ) -> Optional[ValidChunk]:
+
+        # 1. skip magic bytes
+        file.read(4)
+
+        # 2. we parse the frame descriptor of dynamic size
+        flg = self.parse_flg(struct.unpack("<B", file.read(1))[0])
+        logger.debug("Parsed FLG", flg=flg)
```

Comment:
I think this is not helpful, as the `FLG` class does not have a `__repr__` function, so this will print something like `<...FLG object ...>` but nothing more. 

You can add a `__rerp__` for the class, or use `attr` like in `Handler` and `Chunk`. (with `repr=True`)